npm-packages.md

```diff
@@ -1,14 +1,115 @@
+
+> TypeScript files depend on a project-specific configuration, and your setup may not be the same the project that installs your package
+
+Do not include or depend on [polyfills](https://developer.mozilla.org/en-US/docs/Glossary/Polyfill). Leave these decisions to the projects that install your package (see the section on using our packages below).
+
+#### Compiling
+
+Use the TypeScript compiler (`tsc`) to generate the JavaScript files and TypeScript [declaration files](https://www.typescriptlang.org/docs/handbook/declaration-files/introduction.html) that you publish.
+
+Generate two versions of your library:
+
+1. ES modules targetting ES2020
+2. CommonJS modules targetting ES2018 (Node 10+)
+
+> CommonJS is a non-standard module format used by older versions of Node. These files are useful with [Jest](https://jestjs.io/) or if your library is used on the server, for example. At some point we will probably stop including these but currently it adds little overhead and has some value.
+
+##### Example
+
+```js
+// tsconfig.json
+{
+	"compilerOptions": {
+		"declaration": true,
+		"declarationDir": "dist/types",
+		"declarationMap": true,
+		"module": "ES2020",
+		"outDir": "dist/esm",
+		"target": "ES2020"
+	},
+	"files": ["src/index.ts"]
+}
+```
+
+You could then run:
+
+-   `tsc` to generate the default ES modules version and TypeScript declaration files
+-   `tsc --module commonjs --target es2018 --outDir dist/cjs` to generate a CommonJS version
+
+This is only an example, there are many ways of configuring this.
+
+### Configuring your package
+
+NPM packages are described in a `package.json` file.
+
+#### NPM scope
+
+Publish under the [`@guardian`](https://www.npmjs.com/org/guardian) scope.
+
+##### Example
+
+```js
+// package.json
+{
+	"name": "@guardian/my-package",
+	...
+}
+```
+
+#### File references
+
+The ES module version of your library should be referenced in the `module` field.
+
+The CommonJS version should be referenced in the `main` field, and the TypeScript declaration files in the `types` field.
+
+##### Example
+
+```js
+// package.json – based on the `tsconfig.json` above
+{
+	"main": "dist/cjs/index.js",
+	"module": "dist/esm/index.js",
+	"types": "dist/types/index.d.ts",
+	...
+}
+```
+
+### Publishing
+
+#### Continuous delivery
+
+Use [Semantic Release](https://github.com/semantic-release/semantic-release) in a GitHub action with [guardian/actions-merge-release-changes-to-protected-branch](https://github.com/guardian/actions-merge-release-changes-to-protected-branch) to automate publishing.
+
+#### Spontaneous publishing
+
+Publish manually from the command line using [np](https://www.npmjs.com/package/np).
+
+## Using `@guardian` NPM packages
+
+To ensure the Guardian's application bundles are as efficient as possible, packages assume nothing about where they will run.
```

Comment:
that's correct, you can declare a dependency like that, but this refers the environment (browser version etc). i'll try to make to make that clearer

Reply:
Got it 👍

Reply:
That leads me to suggesting adding some recommendations in the **Authoring** section about what to include in `dependencies` vs `peerDependencies`?
One of the problems I recently had was caused by `image-rendering` declaring `@guardian/types` both in `dependencies` and `peerDependencies` with a minor version off. The type errors I was getting from TypeScript by no means were helpful and took me a while to notice that the problem wasn't in the `apps-rendering` codebase!
Just having a quick look at `discussion-rendering` and I can see that there's a babel plugin in its `dependencies` but this seems like a blip.
In `atoms-rendering` there's only a youtube package in its `dependencies` and all the `@guardian` packages live in `peerDependencies` which seems to me like a good call. 
Just mentioning it because if I were to author a new package that depends on `@guardian` packages I'd probably just copy the config from another repo without thinking too much about what package goes where (I probably shouldn't admit to that 🙈).

Reply:
i'm not sure yet _how_ to handle these properly. there seems good arguements to use `peerDependencies` and normal `dependencies` so will leave this for now and come back to it (promise)